specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,429 @@
+Programmatic Save As API
+===
+
+# Background
+
+The context menu has the "Save as" item to manually save the html page, image,
+pdf, or other content through a save as dialog. We provide more flexiable ways 
+to do the save as programmatically in WebView2. You can bring up the default 
+save as dialog easily. And you will be able to block default dialog, save the 
+content silently, by providing the path and save as kind programmatically or 
+even build your own save as UI.
+
+In this document we describe the API. We'd appreciate your feedback.
+
+# Description
+
+We propose the `RequestSaveAs` method WebView2, which allows you to trigger 
+the save as programmatically. By using this method, the system default dialog, 
```

Comment:
"This method launches the system default dialog or the custom UI you implemented by handling the SaveAsRequested event."